backend/runtime/Class.c

```diff
@@ -46,7 +56,13 @@ void Class_destroy(Class *self) {
 }
 
 int64_t Class_fieldIndex(Class *self, String *field) {
-  for (int i = 0; i < self->fieldCount; ++i) {
+  uint64_t initialGuess = String_hash(field) % self->fieldCount;
+  if (String_equals(field, self->fields[initialGuess])) {
+    release(self);
+    release(field);
+    return initialGuess;
+  }
+  for (uint64_t i = inc_mod(initialGuess, self->fieldCount); i != initialGuess; i = inc_mod(i, self->fieldCount)) {
     if (String_equals(field, self->fields[i])) {
```

Comment:
String_equals nie ma w sobie porównania wskaźników, nie wiem czy to ma znaczenie ale tylko Object_equals ma linię   if (self == other) return TRUE;

Mam pewien pomysł jak to później poprawić - na keywordach już jest uniquing, więc jeśli wszędzie zastąpiłbyś stringi keywordami to mógłbyś stosować tylko szybkie porównania "==" (cały uniquing jest już napisany wewnątrz klasy Keyword).

Reply:
Generalnie ja pisałem equals tak, że nigdy nikt nie będzie uruchamiał ręcznie "equals" dla poszczególnych klas tylko zawsze Object_equals. Dlatego np. implementacja keyword_equals jest taka:

`/* outside refcount system */
BOOL Keyword_equals(Keyword *self, Keyword *other) {
  /* What if another thread interns? */
  /* what is here does not matter - keyword equality is checked based on a pointer only */
  return FALSE; //equals(self->string, other->string);
}
`